lm/srilm.py

```diff
@@ -0,0 +1,634 @@
+
+        self.ngram_order = ngram_order
+        self.lm = lm
+        self.vocab = vocab
+        self.eval_data = eval_data
+        self.set_unknown_flag = set_unknown_flag
+        self.ppl_args = ppl_args if ppl_args is not None else ""
+
+        self.ngram_exe = get_ngram_exe(ngram_exe)
+
+        self.rqmt = {
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+            "cpu": cpu_rqmt,
+            "qsub_args": f"-l h_fsize={fs_rqmt}",
+        }
+
+        self.out_ppl_log = self.output_path("perplexity.log", cached=True)
+        self.out_ppl_score = self.output_var("perplexity.score")
+        self.out_num_sentences = self.output_var("num_sentences")
+        self.out_num_words = self.output_var("num_words")
+        self.out_num_oovs = self.output_var("num_oovs")
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt)
+        yield Task("get_ppl", mini_task=True)
+
+    def create_files(self):
+        """creates bash script that will be executed in the run Task"""
+        cmd = [
+            f"{self.ngram_exe.get_path()} \\\n",
+            f"  -order {self.ngram_order} \\\n",
+        ]
+
+        if self.set_unknown_flag:
+            cmd += [f"  -unk \\\n"]
+
+        if self.vocab is not None:
+            cmd += [f"  -vocab {self.vocab.get_cached_path()} \\\n"]
+
+        cmd += [
+            f"  -ppl {self.eval_data.get_cached_path()} \\\n",
+            f"  -lm {self.lm.get_cached_path()} \\\n",
+            f"  {self.ppl_args} &> ppl.log\n",
+        ]
+
+        create_executable("run.sh", cmd)
+
+    def run(self):
+        """executes the previously created script and relinks the log file from work folder to output folder"""
+        self.sh("./run.sh")
+        relink("ppl.log", self.out_ppl_log.get_path())
+
+    def get_ppl(self):
+        """extracts various outputs from the ppl.log file"""
+        with open(self.out_ppl_log.get_cached_path(), "rt") as f:
+            lines = f.readlines()[-2:]
+            for line in lines:
+                line = line.split(" ")
+                for idx, ln in enumerate(line):
+                    if ln == "sentences,":
+                        self.out_num_sentences.set(int(line[idx - 1]))
+                    if ln == "words,":
+                        self.out_num_words.set(int(line[idx - 1]))
+                    if ln == "OOVs":
+                        self.out_num_oovs.set(int(line[idx - 1]))
+                    if ln == "ppl=":
+                        self.out_ppl_score.set(float(line[idx + 1]))
+
+    @classmethod
+    def hash(cls, kwargs):
+        """delete the queue requirements from the hashing"""
+        del kwargs["mem_rqmt"]
+        del kwargs["cpu_rqmt"]
+        del kwargs["time_rqmt"]
+        del kwargs["fs_rqmt"]
+        return super().hash(kwargs)
+
+
+class ComputeBestMixJob(Job):
+    """
+    Compute the best mixture weights from given PPL logs
```

Comment:
```suggestion
    Compute the best mixture weights for a combination of count LMs based on the given PPL logs
```
maybe we can also add "linear" or "log-linear" if we know what actually happens here xD